python/tidy/linting_report.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2025 The Servo Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+
+from typing import List, Literal, NotRequired, TypedDict
+
+import colorama
+
+
+class GithubAnnotation(TypedDict):
+    file_name: str
+    line_start: int
+    line_end: int
+    level: Literal["notice", "warning", "error"]
+    title: str
+    message: str
+    column_start: NotRequired[int]
+    column_end: NotRequired[int]
+
+
+class LintingReportManager:
+    def __init__(self, annotation_prefix: str, limit: int):
+        self.annotation_prefix: str = annotation_prefix
+        self.limit: int = limit
+        self.severenty_map: dict[str, Literal["notice", "warning", "error"]] = {
+            "help": "notice",
+            "note": "notice",
+            "warning": "warning",
+            "error": "error",
+        }
+        self.annotations: List[GithubAnnotation] = []
+        self.total_count = 0
+        colorama.init()
+
+    def error_log(self, error):
+        print(
+            "\r  | "
+            + f"{colorama.Fore.BLUE}{error[0]}{colorama.Style.RESET_ALL}:"
+            + f"{colorama.Fore.YELLOW}{error[1]}{colorama.Style.RESET_ALL}: "
+            + f"{colorama.Fore.RED}{error[2]}{colorama.Style.RESET_ALL}"
+        )
+
+    def clean_path(self, path):
+        return path.removeprefix("./")
+
+    def escape(self, s):
+        return s.replace("\r", "%0D").replace("\n", "%0A")
+
+    def append_annotation(
+        self,
+        title,
+        message,
+        file_name,
+        line_start,
+        line_end=None,
+        annotation_level=None,
+        column_start=None,
+        column_end=None,
+    ):
+        if self.total_count >= self.limit:
+            return
+
+        if annotation_level is None:
+            annotation_level = "error"
+
+        if line_end is None:
+            line_end = line_start
+
+        annotation: GithubAnnotation = {
+            "title": f"./Mach {self.annotation_prefix}: {title}",
```

Comment:
```suggestion
            "title": f"./mach {self.annotation_prefix}: {title}",
```

I would suggest changing `./Mach` to `./mach` so that it matches the CLI invocation. However, maybe just `clippy` or 'test-tidy' would also be sufficient. Maybe we can discuss this in our call today.